File: onto-code-maker-result-tester/src/test/python/familyPython/serialization/CatSerialization.py

from rdflib.graph import URIRef, Literal,BNode,Collection
from rdflib.namespace import RDF
from Vocabulary import Vocabulary
from entities.Cat import Cat
from .SerializationModel import SerializationModel

class CatSerialization(SerializationModel):

    def addToModel(self,ontology, cat):
        ontology.add((cat.getIri(),RDF.type, cat.getClassIRI()))
        self.addPropertiesToModel(ontology,cat)


    def addPropertiesToModel(self,ontology, cat):
        pass

    def setProperties(self,ontology, cat, nestingLevel):
        pass

    def getInstanceFromModel(self,ontology, instanceIri,nestingLevel):
        if (instanceIri,RDF.type,Cat.CLASS_IRI) in ontology:
            cat =  Cat(instanceIri)
            if nestingLevel > 0:
                nestingLevel -= 1
                self.setProperties(ontology, cat,nestingLevel)
            return cat
        return None

    def getAllInstancesFromModel(self,ontology,nestingLevel):
        allInstances = set()
        for s, p, o in ontology.triples((None,RDF.type,Cat.CLASS_IRI)):
            if isinstance(s,URIRef) :
                cat = self.getInstanceFromModel(ontology,s,nestingLevel)
                allInstances.add(cat)
        return allInstances

    def removeInstanceFromModel(self,ontology, subject):
        ontology.remove((subject,RDF.type,Cat.CLASS_IRI))
        for statement in ontology.triples((subject,None,None)):
            ontology.remove(statement)

File: onto-code-maker-result-tester/src/main/java/ontology/generator/classes/examples/familySimple/serialization/CatSerialization.java

package ontology.generator.classes.examples.familySimple.serialization;

import org.eclipse.rdf4j.model.*;
import java.util.Collection;
import java.util.Set;
import java.util.HashSet;
import java.util.*;

import java.util.stream.Collectors;
import ontology.generator.classes.examples.familySimple.entities.*;
import org.eclipse.rdf4j.model.util.Values;
import org.eclipse.rdf4j.model.vocabulary.RDF;

import java.time.LocalTime;
import ontology.generator.classes.examples.familySimple.Vocabulary;

public class CatSerialization extends SerializationModel<Cat>{

    @Override
    public void addToModel(Model model, Cat cat) {
        model.add(cat.getIri(),RDF.TYPE, cat.getClassIRI());
        addPropertiesToModel(model,cat);

    }

    protected void addPropertiesToModel(Model model,  Cat cat) {

    }

    protected void setProperties(Model model,Cat cat,int nestingLevel) throws Exception{

    }

    @Override
    public Cat getInstanceFromModel(Model model,IRI instanceIri,int nestingLevel) throws Exception{
        Model statements = model.filter(instanceIri,RDF.TYPE,Cat.CLASS_IRI);
        if(statements.size() != 0){
            Cat cat = new Cat(instanceIri);
            if(nestingLevel > 0){
                nestingLevel--;
                setProperties(model, cat,nestingLevel);
            }
            return cat;
        }

        return null;
    }

    @Override
    public Collection<Cat> getAllInstancesFromModel(Model model,int nestingLevel)throws Exception{
        Model statements = model.filter(null,RDF.TYPE,Cat.CLASS_IRI);
        Collection<Cat> allInstances = new ArrayList<>();
        for(Statement statement:statements){
            Resource subject = statement.getSubject();
            if(subject.isIRI()){
                IRI iri = (IRI) subject;
                Cat cat = getInstanceFromModel(model,iri,nestingLevel);
                allInstances.add(cat);
            }
        }

        return allInstances;
    }

    @Override
    public void removeInstanceFromModel(Model model,IRI instanceIri) {
        Set<Value> rdfCollections = model.filter(instanceIri,null,null).objects();
        List<Value> listOfnodes = rdfCollections.stream().filter(o -> o.isBNode()).collect(Collectors.toList());
        for(Value node:listOfnodes){
            model.removeAll(getModelRDFCollection(model,(BNode)node));
            model.remove(instanceIri,null,(BNode)node);
        }
        model.remove(instanceIri,RDF.TYPE,Cat.CLASS_IRI);
        model.remove(instanceIri,null,null);
    }

    @Override
    public void updateInstanceInModel(Model model,Cat cat){
        Set<Value> rdfCollections = model.filter(cat.getIri(),null,null).objects();
        List<Value> listOfnodes = rdfCollections.stream().filter(Value::isBNode).collect(Collectors.toList());
        for(Value node:listOfnodes){
            model.removeAll(getModelRDFCollection(model,(BNode)node));
            model.remove(cat.getIri(),null,node);
        }

        Model statements = model.filter(cat.getIri(),null,null);
        statements.removeIf(x -> !x.getPredicate().equals(RDF.TYPE));

        addPropertiesToModel(model,cat);
    }



}

